packages/flutter/lib/src/widgets/form.dart

```diff
@@ -291,18 +291,46 @@ class FormState extends State<Form> {
   /// returns true if there are no errors.
   ///
   /// The form will rebuild to report the results.
+  ///
+  /// See also:
+  ///  * [validateGranularly], which also validates descendant [FormField]s,
+  /// but instead returns a [Set] of fields with errors
   bool validate() {
     _hasInteractedByUser = true;
     _forceRebuild();
     return _validate();
   }
 
-  bool _validate() {
+
+  /// Validates every [FormField] that is a descendant of this [Form], and
+  /// returns a [Set] of [FormFieldState] of the invalid field(s) only, if any.
+  ///
+  /// This method can be useful to highlight field(s) with errors,
+  /// possibly using their widget key(s).
```

Comment:
Looks like the documentation here needs updating?

Reply:
It still holds IMO. The widget keys are still most likely what the user will use to highlight fields with errors on the screen.

Reply:
If you have the states you don't need keys no?

Reply:
^^ maybe this can just be `This method can be useful to highlight field(s) with errors.`